custom_components/ha-google-home/sensor.py

```diff
@@ -1,38 +1,53 @@
-"""Sensor platform for Google local authentication token fetching."""
+"""Sensor platforms for Google Home"""
+import logging
+
 from homeassistant.const import STATE_OFF
 from homeassistant.const import STATE_ON
 
 from .const import DOMAIN
 from .const import FIRE_TIME_IN_S
+from .const import LABEL_ALARMS
+from .const import LABEL_TIMERS
+from .const import LABEL_TOKEN
 from .entity import GoogleHomeAlarmEntity
-from .entity import GoogleHomeEntity
 from .entity import GoogleHomeTimersEntity
+from .entity import GoogleHomeTokenEntity
 from .utils import format_alarm_information
 from .utils import format_timer_information
 
 
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
 async def async_setup_entry(hass, entry, async_add_devices):
     """Setup sensor platform."""
     coordinator = hass.data[DOMAIN][entry.entry_id]
-    for device in coordinator.data:
-        if device.local_auth_token:
-            async_add_devices(
-                [
-                    GoogleHomeAlarmSensor(coordinator, entry, device),
-                    GoogleHomeTimerSensor(coordinator, entry, device),
-                    GoogleHomeSensor(coordinator, entry, device),
-                ]
+    for device_name, device_data in coordinator.data.items():
+        devices_to_add = []
+        if device_data[LABEL_TOKEN]:
+            devices_to_add.append(
+                GoogleHomeTokenSensor(coordinator, entry, device_name, device_data)
+            )
+        if device_data[LABEL_ALARMS]:
+            devices_to_add.append(
+                GoogleHomeAlarmSensor(coordinator, entry, device_name, device_data)
+            )
+        if device_data[LABEL_TIMERS]:
+            devices_to_add.append(
+                GoogleHomeTimerSensor(coordinator, entry, device_name, device_data)
             )
+        if devices_to_add:
+            async_add_devices(devices_to_add)
```

Comment:
This basically means if there is no alarm or timer when platform is setup (home assistant started) then corresponding device won't be created and it won't get any updates. You should probably revert this.